jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java

```diff
@@ -79,4 +96,79 @@ private static String getWorkspaceInvisibleProjectName(IPath workspacePath) {
         String fileName = workspacePath.toFile().getName();
         return fileName + "_" + Integer.toHexString(workspacePath.toPortableString().hashCode());
     }
+
+    public static List<PackageNode> getMainMethod(IProgressMonitor monitor) {
+        final List<PackageNode> res = new ArrayList<>();
+        IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+        SearchPattern pattern = SearchPattern.createPattern("main(String[]) void", IJavaSearchConstants.METHOD,
+                IJavaSearchConstants.DECLARATIONS, SearchPattern.R_EXACT_MATCH| SearchPattern.R_CASE_SENSITIVE);
+        SearchRequestor requestor = new SearchRequestor() {
+            @Override
+            public void acceptSearchMatch(SearchMatch match) {
+                Object element = match.getElement();
+                if (element instanceof IMethod) {
+                    IMethod method = (IMethod) element;
+                    try {
+                        if (method.isMainMethod()) {
```

Comment:
Get it and will be fixed in the next commit.

Reply:
> Too deep if hierarchy here.
> 
> If you want to do the null check, please try:
> 
> ```java
> if (for == null) {
>     return bar;
> }
> ```
> 
> something like that. The purpose is to avoid deep if calls.

Get it and will be fixed in the next commit.